### ISSUE TITLE
OnePlus 7T imei number not being spoofed

### ISSUE THREAD

**Issue by Raj2032** [Android1500/AndroidFaker#66]
I have rooted my device, installed magisk, lsposed and enabled module system settings, rebooted device and put in spoofed imei number.

I typed on my dial pad `*#06#` and it still shows original imei number, not spoofed imei number.

**Comment by FluffyFox337** [comment]
> Android Faker does NOT fake your real IMEI, this is explained clearly in the readme.
> It does only spoof values in other apps.
> Also, don't use the latest version, it's completely broken- stay with 1.9.2.
> 
> To properly use it, do as you already did, but then select the apps you want to spoof inside LSPosed manually.
> Also, select the values you want to have spoofed inside the app itself.
> 

Good afternoon. Can you please tell me if I understand correctly that this program does not spoof MAC for the router and other external devices in the local network? 
 The router will still see my real MAC?

**Comment by Raj2032** [comment]
@FluffyFox337 It appears that way.

**Comment by Raj2032** [comment]
> BUT I wouldn't try this if I were you.

@HarriBuh are you referring to using the app or like trying to spoof the imei number for cell towers?

**Comment by FluffyFox337** [comment]
> [@FluffyFox337](https://github.com/FluffyFox337) I can only say that it spoofs values inside apps. Routers and other hardware are running on software too after all, BUT I wouldn't try this if I were you.

thanks for reply. I made sh script for change MAC on phones with root rights. Temporary MAC active only while one connection session. So, script automatically change MAC after his reset. If you wanna, i can create repository with that script.

**Comment by Raj2032** [comment]
@FluffyFox337 Can you advise me can your script also change IMEI number as well?

**Comment by FluffyFox337** [comment]
> [@FluffyFox337](https://github.com/FluffyFox337) Can you advise me can your script also change IMEI number as well?

I have not tried to write a script to change imei. I can only assume that it can also be virtually temporarily changed until the phone is rebooted. On older phones it was possible to change imei which was loaded from read only partition to data partition and was active until factory reset when original imei was loaded again. I didn't go into details of how newer android versions work.

**Comment by FluffyFox337** [comment]
> [@FluffyFox337](https://github.com/FluffyFox337) Can you advise me can your script also change IMEI number as well?

I also believe that the imei can be changed via e.g. twrp by flashing a modified backup of the imei info section. (efs, etc.)

**Comment by Raj2032** [comment]
@FluffyFox337 yes you can't actually modify imei nuimber as it is burned in address however I mean temporarily change. Like how you can spoof your wifi mac address.

> I also believe that the imei can be changed via e.g. twrp by flashing a modified backup of the imei info section. (efs, etc.)

That could be helpful but can you give me more info where I need to go and what I need to flash and which filepath?
